lxd/storage/backend_lxd.go

```diff
@@ -3952,7 +3952,15 @@ func (b *lxdBackend) CreateCustomVolume(projectName string, volName string, desc
 		return err
 	}
 
-	b.state.Events.SendLifecycle(projectName, lifecycle.StorageVolumeCreated.Event(vol, string(vol.Type()), projectName, op, logger.Ctx{"type": vol.Type()}))
+	eventCtx := logger.Ctx{"type": vol.Type()}
+	if !b.Driver().Info().Remote {
+		node, ok := b.nodes[b.state.DB.Cluster.GetNodeID()]
+		if ok {
+			eventCtx["location"] = node.Name
+		}
+	}
+
+	b.state.Events.SendLifecycle(projectName, lifecycle.StorageVolumeCreated.Event(vol, string(vol.Type()), projectName, op, eventCtx))
```

Comment:
@tomponline I'm not sure if this is the best way to get the node name here. Also I wonder if this will fail when LXD is not clustered. Double checking tests now and will mark as draft until I'm sure.

Reply:
I'm taking a look now, I think we can do it slightly cheaper.

Reply:
Should be able to use s.ServerName

Reply:
Ah excellent thank you